pkg/controller/pxcbackup/controller.go

```diff
@@ -252,18 +252,27 @@ func (r *ReconcilePerconaXtraDBClusterBackup) Reconcile(request reconcile.Reques
 	return rr, err
 }
 
-func removeS3Finalizer(cr *api.PerconaXtraDBClusterBackup) {
-	filteredFins := make([]string, 0)
-
-	for _, f := range cr.GetFinalizers() {
-		if f == api.FinalizerDeleteS3Backup {
-			continue
+func removeS3Finalizer(cl client.Client, cr *api.PerconaXtraDBClusterBackup) error {
+	finalizers := cr.GetFinalizers()
+	beforeLen := len(finalizers)
+
+	for i := 0; i != len(finalizers); i++ {
+		if finalizers[i] == api.FinalizerDeleteS3Backup {
```

Comment:
I think the gains from the optimization don't compensate the loss of readability

Reply:
but this is idiomatic way to remove item from slice